src/InputField.js

```diff
@@ -9,72 +9,144 @@ import Label from './Label'
-          width={1}
-          mb={-24}
-          color={theme.colors[color] || theme.colors.borderGray}
-          bg={theme.colors[color] || theme.colors.borderGray}
-        >
-          <Text
-            id={`${id}-info`}
-            bold
-            fontSize={10}
-            px={13}
-            py={1}
-            color={'white'}
-          >
-            {info}
-          </Text>
-        </InfoBox>
-      )}
-    </div>
-  )
+    let InputChild
+    let inputPosition = -1
+    let LabelChild
+    let BeforeIcon
+    let AfterIcon
+    let inputId
+    let iconAdjustment
+
+    React.Children.forEach(children, (child, index) => {
+      if (child) {
+        if (child.type === Label) {
+          LabelChild = child
+        }
+        if (child.type === Input) {
+          inputPosition = index
+          InputChild = child
+          inputId = child.props.id
+        }
+        if (child.type === Icon) {
+          if (inputPosition < 0) {
+            BeforeIcon = child
+            iconAdjustment = child.props.size - Icon.defaultProps.size
+          } else {
+            AfterIcon = child
+          }
+        }
+      }
+    })
+
+    // Handle old version on component's api
+    if (icon) {
+      AfterIcon = <Icon name={icon} />
+    }
+    if (label) {
+      LabelChild = <Label>{label}</Label>
+    }
+    if (!InputChild) {
+      InputChild = <Input />
+    }
+
+    return (
+      <Box>
+        {LabelChild &&
+          React.cloneElement(LabelChild, {
+            pl: BeforeIcon ? 40 : 2,
+            mt: '6',
+            htmlFor: inputId
+          })}
+        <Flex align="center" width={1} mt={0}>
+          {BeforeIcon && (
+            <Box mr={-4} ml={8 - iconAdjustment} mt={LabelChild ? -12 : 1}>
+              {BeforeIcon}
+            </Box>
+          )}
+          {React.cloneElement(InputChild, {
+            pt: LabelChild ? 20 : 14,
+            pb: LabelChild ? 8 : 14,
+            mt: LabelChild && -20,
+            pr: AfterIcon && 40,
+            pl: BeforeIcon ? 40 : 2,
+            width: 1,
+            color: color,
+            ...props
+          })}
+          {AfterIcon && (
+            <Box ml={-4} mt={LabelChild ? -12 : 1}>
+              {AfterIcon}
+            </Box>
+          )}
+        </Flex>
+      </Box>
+    )
+  }
 }
 
 InputField.propTypes = {
-  /** Id passed to the native input element. Necessary to set up aria-describedby and for attributes for accessibility.  */
-  id: PropTypes.string.isRequired,
-  /** Text to be used as the label for the input element. */
-  label: PropTypes.string.isRequired,
-  /** The optional icon that shows up on the right side of the element. */
-  icon: PropTypes.string,
-  /** Optional border and icon color. */
-  color: PropTypes.string,
-  /** Additional content to go in the info drawer below the element. This will be replaced by a tooltip later probably.  */
-  info: PropTypes.string
+  children: function(props, propName, componentName) {
```

Comment:
This prop type checking may be overkill, but since this component does still have a pretty rigid API I thought it worth doing.